Update readme: new feature set (#23)
doctolib/safe-pg-migrations: README.md

- Long-running queries are active on the table. To create / remove an index, PG needs to wait for the queries that are actually running to finish before starting the index creation / removal. The blocking activity logger might help you to pinpoint the culprit queries.
- A vacuum / autovacuum is running on the table, holding a ShareUpdateExclusiveLock, you are most likely out of luck for the current migration, but you may try to [optimize your autovacuums settings](https://www.percona.com/blog/2018/08/10/tuning-autovacuum-in-postgresql-and-autovacuum-internals/).

</details>

<details><summary>safe <code>add_foreign_key</code> (and <code>add_reference</code>)</summary>

> Adding a foreign key requires by default a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.

Reviewer: ```suggestion
Adding a foreign key requires a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.
```

> Adding the constraint is rather fast, the major part of the migration is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key is safe by splitting the addition of the foreign key in two parts: 

Reviewer: ```suggestion
Adding the constraint itself is rather fast, the major part of the time is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key holds blocking locks for the least amount of time by splitting the foreign key creation in two steps: 
```

> 1. adding the constraint *without validation*;

Reviewer: ```suggestion
1. adding the constraint *without validation*, will not validate existing rows;
```

> 2. validating the constraint.  

Reviewer: ```suggestion
2. validating the constraint, will validate existing rows in the table, without blocking read or write on the table
```

> Indeed the validation of a constraint does not require any lock preventing reading or writing on the tables.

Reviewer: I'd remove this one then

Author: ```suggestion
```
